File: engine/effects/handlers/damage.py
"""Damage effect handler scaffold."""

from __future__ import annotations

from engine.effects.base import Effect


def handle_damage(effect: Effect) -> dict:
    return {
        "status": "ok",
        "kind": "damage",
        "target": effect.target_unit_id,
        "payload": effect.payload,
    }

File: src/effects/handlers/damage.ts
/**
 * Damage effect handler scaffold.
 * Mirrors engine/effects/handlers/damage.py
 */

import { Effect } from "../base";

export function handleDamage(effect: Effect): Record<string, unknown> {
  return {
    status: "ok",
    kind: "damage",
    target: effect.targetUnitId,
    payload: effect.payload,
  };
}
